pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4085,9 +4085,21 @@ public static CompletableFuture<PartitionedTopicMetadata> unsafeGetPartitionedTo
                 }
                 metadataFuture.complete(metadata);
             }).exceptionally(ex -> {
-            metadataFuture.completeExceptionally(ex.getCause());
-            return null;
-        });
+                    if (ex.getCause() instanceof MetadataStoreException.AlreadyExistsException) {
+                        // The partitioned topic might be created concurrently
+                        pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(topicName)
```

Comment:
Why don't we move this exception capture to here: https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L2642-L2645

Reply:
Good suggestions. I was concerned about the effect on other places before. But it should be the right behavior for it, including the admin operations.